src/main/java/Group15/Util/WorkoutAlgorithm.java

```diff
@@ -92,7 +92,7 @@ private static Exercise getValidExercise(List<BodyPart> selectedBodyParts, List<
             {
                 if (exercise.bodyParts.contains(bodyPart))
                 {
-                    continue;
+                    break;
```

Comment:
This seems to work, but i have had the issue that it just cycles back and forth between 2 muscles. To recreate i chose the first 4 muscles, with equipments: Barbell, Dumbbell, Cable and Machine. All i got was Bicep and Chest exercises, no Abdominal exercises or Calves.

Reply:
Can't reproduce, so im marking it as resolved

Reply:
bro XD